cmd/peer-rest-server.go

```diff
@@ -1137,31 +1137,8 @@ type SpeedtestResult struct {
 	Error     string
 }
 
-// SpeedtestObject implements "random-read" object reader
-type SpeedtestObject struct {
-	buf               []byte
-	remaining         int
-	totalBytesWritten *uint64
-}
-
-func (bo *SpeedtestObject) Read(b []byte) (int, error) {
-	if bo.remaining == 0 {
-		return 0, io.EOF
-	}
-	if len(b) == 0 {
-		return 0, nil
-	}
-	if len(b) > len(bo.buf) {
-		b = b[:len(bo.buf)]
-	}
-	if len(b) > bo.remaining {
-		b = b[:bo.remaining]
-	}
-	copy(b, bo.buf)
-	bo.remaining -= len(b)
-
-	atomic.AddUint64(bo.totalBytesWritten, uint64(len(b)))
-	return len(b), nil
+func newRandomReader(size int) io.Reader {
+	return io.LimitReader(rand.New(rand.NewSource(time.Now().UnixNano())), int64(size))
```

Comment:
Last time I checked this was very CPU intensive. Check this is a reasonable concurrency. If it hasn't changed the random function is quite slow and a clear bottleneck.

Reply:
okay @klauspost will check

Reply:
@harshavardhana Here is a random source, that is 10x the `math/rand` speed: https://github.com/minio/pkg/pull/20